tests/lib/store.sh

```diff
@@ -15,57 +19,45 @@ EOF
     systemctl start snapd.socket
 }
 
+setup_staging_store(){
+    _configure_store_backends "SNAPPY_USE_STAGING_STORE=1"
+}
+
+teardown_staging_store(){
+    if systemctl status snapd.socket; then
+        systemctl stop snapd.socket
+    fi
+    rm -rf $STORE_CONFIG
+    systemctl daemon-reload
+    systemctl start snapd.socket
+}
+
 setup_fake_store(){
     local top_dir=$1
-    shift
+
     mkdir -p $top_dir/asserts
     # debugging
     systemctl status fakestore || true
     echo "Given a controlled store service is up"
 
     https_proxy=${https_proxy:-}
     http_proxy=${http_proxy:-}
-    systemd_create_and_start_unit fakestore "$(which fakestore) -start -dir $top_dir -addr localhost:11028 -https-proxy=${https_proxy} -http-proxy=${http_proxy} $@" "SNAPD_DEBUG=1 SNAPD_DEBUG_HTTP=7"
+    systemd_create_and_start_unit fakestore "$(which fakestore) -start -dir $top_dir -addr localhost:11028 -https-proxy=${https_proxy} -http-proxy=${http_proxy} -assert-fallback" "SNAPD_DEBUG=1 SNAPD_DEBUG_HTTP=7"
```

Comment:
here when we run fakestore we also need to pass to it SNAPPY_USE_STAGING_STORE=1 if we are using staging